File: Flink/src/main/scala/Testing/WordCount.scala
package Testing

import org.apache.flink.streaming.api.scala._

/**
 * @author wufc
 * @create 2020-04-05 11:37 上午
 */
object WordCount {
  def main(args: Array[String]): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    env.socketTextStream("localhost",9999)
      .map(line=>(line,1)).uid("a")
      .keyBy(_._1)
      .sum(1)
  }

}

File: Flink/src/test/java/WordCount.java
import org.apache.flink.api.common.functions.FlatMapFunction;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.util.Collector;

/**
 * @author wufc
 * @create 2020-04-17 10:31 下午
 */
public class WordCount {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);

        env.socketTextStream("localhost",9999)
                .flatMap(new FlatMapFunction<String, Tuple2<String,Integer>>() {
                    @Override
                    public void flatMap(String value, Collector<Tuple2<String, Integer>> out) throws Exception {
                        String[] arr = value.split(",", -1);
                        for (String s : arr) {
                            out.collect(Tuple2.of(s,1));
                        }
                    }
                })
                .keyBy(0)
                .sum(1)
                .print();

        env.execute("Test");
    }
}
